src/main/java/spiffe/api/svid/util/ExponentialBackOff.java

```diff
@@ -0,0 +1,75 @@
+package spiffe.api.svid.util;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * Implements a basic Exponential Backoff Policy
+ * Retries based on a List of Errors
+ *
+ */
+public class ExponentialBackOff {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(ExponentialBackOff.class);
+
+    static int MAX_RETRIES = 5;
+    static int BASE = 2;
+
+    /**
+     * The list of Errors that can produce the WorkloadApi and must cause a Retry
+     */
+    private static List<String> RETRYABLE_ERRORS = Arrays.asList("UNAVAILABLE", "PERMISSIONDENIED");
```

Comment:
What about errors related to the socket not being found? This is likely to be encountered during agent upgrade or similar action. Perhaps timeout is also something to consider here...? We definitely don't want this to be fragile - it should survive basically anything. We can assume that if the workload API client doesn't survive, then neither will the workload since it needs the identity provided by it in order to complete its work.